### PR TITLE
MOSTLY fixes gym auto selects provisioning profile mapping wrong in an Xcode project with multiple configs

### LINKED ISSUE #9890: Gym auto selects provisioning profile mapping wrong in an Xcode project with multiple configs
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

Running the lane `deploy_testflight`, gym generate a plist file with the wrong provisioning profile 
```
[13:27:07]: Generated plist file with the following values:
[13:27:07]: ▸ -----------------------------------------
[13:27:07]: ▸ {
[13:27:07]: ▸   "provisioningProfiles": {
[13:27:07]: ▸     "com.xxx.yyy${BUNDLE_ID_SUFFIX}": "xxx Dev AdHoc Distribution"
[13:27:07]: ▸   },
[13:27:07]: ▸   "method": "app-store"
[13:27:07]: ▸ }
[13:27:07]: ▸ -----------------------------------------
```
`BUNDLE_ID_SUFFIX` is a user define config variabile I have in my project.
It's value, for a TestFlight build (`ProdRelease` config) is empty btw.

The problem is that the chosen provisioning profile is wrong. To deploy the app on TestFlight, the AppStore distribution profile must be used.
The Xcode project is correctly configured to do so

---

Edit by @joshdholtz: This method is the causing this issue when Xcode project have multiple configurations - https://github.com/fastlane/fastlane/blob/6a82b16373568b0d2cd793998dd6164503831aef/gym/lib/gym/detect_values.rb#L89-L120. It sometimes selects the wrong profile and makes things go 💥 

---

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.5                                     |
| Ruby                        | 2.4.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.11.0                          |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.12.5 (16F73)                    |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.4.0/lib            |
| OpenSSL Version             | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.1                                       |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
xcode_select "/Applications/Xcode.app"

# Fastlane constants
fastlane_version "2.7.0"
default_platform :ios
itc_uri = "https://itunesconnect.apple.com/WebObjects/iTunesConnect.woa/ra/ng/app/" + "xyz"

platform :ios do

  before_all do
    
    # Shared information
    ENV['APP_MARKETING_VERSION'] = `cd .. && agvtool what-marketing-version -terse1 | tr -d '\n'`

  end

  desc "Build and deploy app for public testing on TestFlight"
  lane :deploy_testflight do
    
    # Check ENV
    raise "Please specify SLACK_URL inside `.env` file before run this lane again".red unless ENV['SLACK_URL']
    raise "Please specify SLACK_USERNAME inside `.env` file before run this lane again".red unless ENV['SLACK_USERNAME']
    
    # Ensure what-to-test is up-to-date
    UI.header "testflight-what-to-test.txt"
    UI.message(File.read("./metadata/testflight-what-to-test.txt"))
    unless UI.confirm("Is the what-to-test correct?")
      UI.user_error!("Please update `testflight-what-to-test.txt` file.")
    end

    # Build
    build_ipa(environment: 'prod')

    # Upload and share build with external betatesters
    upload_ipa_testflight

    # Notify on Slack
    slack({
      message: "*Radio iOS #{ENV['APP_MARKETING_VERSION']}* has been uploaded and it's waiting for betatesting review 🚀",
      success: true,
      username: "#{ENV['SLACK_USERNAME']} (fastlane)",
      channel: '#releases',
      payload: {
        'CHANGELOG' => "```" + File.read("./metadata/testflight-what-to-test.txt") + "```"
        },
      default_payloads: [], # No default payload
      })

  end

  desc "Build ipa for AppStore or Ad-Hoc distribution"
  private_lane :build_ipa do |options|

    # Check options
    scheme = ""
    output = ""
    configuration = ""
    export = ""

    case options[:environment]
    when 'prod'
      scheme = "Radio Prod"
      output = "Radio Prod"
      configuration = "ProdRelease"
      export = "app-store"
    
    when 'dev'
      scheme = "Radio Dev"
      output = "Radio Dev"
      configuration = "DevRelease"
      export = "ad-hoc"
    
    when 'beta'
      scheme = "Radio Beta"
      output = "Radio Beta"
      configuration = "BetaRelease"
      export = "ad-hoc"
    
    else
      UI.user_error!("Invalid environment passed to lane: #{options[:environment]}")
    end

    UI.message("Building scheme `#{scheme}` with configuration `#{configuration}`")

    # pod install
    cocoapods

    # Build
    gym(
      scheme: scheme,
      output_name: output,
      configuration: configuration,
      export_method: export
      )

  end

  desc "Upload ipa (PROD) to TestFlight and HockeyApp"
  private_lane :upload_ipa_testflight do

    # Check ENV
    raise "Please specify HOCKEYAPP_TOKEN inside `.env` file before run this lane again".red unless ENV['HOCKEYAPP_TOKEN']
    raise "Please specify SLACK_URL inside `.env` file before run this lane again".red unless ENV['SLACK_URL']
    raise "Please specify SLACK_USERNAME inside `.env` file before run this lane again".red unless ENV['SLACK_USERNAME']

    # Upload build to iTunesConnect
    pilot(
      username: ENV['FASTLANE_USERNAME'],
      changelog: File.read("./metadata/testflight-what-to-test.txt"),
      beta_app_description: File.read("./metadata/testflight-description.txt"),
      beta_app_feedback_email: "xxx@yyy.com",
      distribute_external: true,
      skip_submission: true
      )

    # Send dSYM to HockeyApp
    upload_ipa_hockeyapp(
      environment: 'prod',
      notes: "New build available in AppStore for external testers."
      )

    # Notify on Slack
    slack({
      message: "Uploaded Radio iOS (PROD) app #{ENV['APP_MARKETING_VERSION']} to TestFlight",
      success: true,
      channel: '#bot',
      username: "#{ENV['SLACK_USERNAME']} (fastlane)",
      payload: {
        'iTunes Connect link' => itc_uri
        },
      default_payloads: [], # No default payload
      })

  end

  desc "Upload ipa to HockeyApp"
  private_lane :upload_ipa_hockeyapp do |options|

    raise "Please specify `environment` parameter".red unless options[:environment]

    identifier = ""
    dsym_only = false
    notes = options[:notes] || "New build available on HockeyApp."

    case options[:environment]
    when 'prod'
      identifier = '123'
      dsym_only = true
    
    when 'dev'
      identifier = '456'
      dsym_only = false
    
    when 'beta'
      identifier = '789'
      dsym_only = false
    
    else
      UI.user_error!("Invalid environment passed to lane: #{options[:environment]}")
    end

    UI.message("Uploading dsym only? #{dsym_only}")

    hockey(
      api_token: ENV['HOCKEYAPP_TOKEN'],
      upload_dsym_only: dsym_only,
      public_identifier: identifier,
      notes: notes
      )

  end

  #
  # On lane success
  #
  after_all do |lane|

    unless is_ci?
      # Mac OS notification
      notification(
        title: "Radio iOS 🚀: 👍", 
        message: "#{lane} succeded!"
        )
    end

  end

  #
  # On lane failure
  #
  error do |lane, exception|

    unless is_ci?
      # Mac OS notification
      notification(
        title: "Radio 🚀: 🔥", 
        message: "#{lane} failed: #{exception.message}"
        )
    end

  end
end

# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/docs/Actions.md
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# More information: https://github.com/fastlane/fastlane/blob/master/docs/Appfile.md

for_lane :dev do
  app_identifier "com.xxx.yyy.dev"
  app_icon "./Resources/AppIcon-dev-1024.png"
end

for_lane :beta do
  app_identifier "com.xxx.yyy.beta"
  app_icon "./Resources/AppIcon-beta-1024.png"
end
```
</details>

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.47.0  | 🚫 Update available |

### Loaded fastlane plugins:

| Plugin                 | Version | Update-Status |
| ---------------------- | ------- | ------------- |
| fastlane-plugin-framer | 0.2.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| httpclient                | 2.8.3        |
| google-api-client         | 0.12.0       |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| bundler                   | 1.14.3       |
| faraday_middleware        | 0.11.0.1     |
| json                      | 2.1.0        |
| io-console                | 0.4.6        |
| excon                     | 0.57.1       |
| openssl                   | 2.0.2        |
| plist                     | 3.3.0        |
| faraday                   | 0.12.2       |
| jwt                       | 1.5.6        |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.3        |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| terminal-table            | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| fastlane-plugin-framer    | 0.2.0        |
</details>

*generated on:* **2017-07-28**
</details>

### PR DESCRIPTION
**Pull request by joshdholtz**
Fixes #9890

^^ Or at least helps to fix this 😇 

### Problem
`gym` will create the `export_options.provisionalProfiles` mapping wrong when:
1) There were multiple configurations with the same bundle identifier 
2) `xcconfig` files are being used to specific the `PRODUCT_BUNDLE_IDENTIFIER`, `PROVISIONING_PROFILE_SPECIFIER`, and/or `PROVISIONING_PROFILE`

This was resulting in a mapping with keys of `nil` (due to `PRODUCT_BUNDLE_IDENTIFIER`) not being resolved from the `xcconfig` file and/or the wrong provisioning profile selected due to duplicate bundle identifiers that are overwriting previously mapped provisioning profiles

#### xcconfig file
To solve the `xcconfig` files from not being resolved, I upgraded the `xcodeproj` gem from at least `1.4.4` to at least `1.5.0`. `1.5.0` [added a method](https://github.com/CocoaPods/Xcodeproj/releases/tag/1.5.0) called `resolve_build_setting` which takes into account any referenced `xcconfig` files. This will now solve the `nil` key that appears when a bundle identifier is specified via a `xcconfig` file`

#### duplicate bundle identifier
This issue is not 100% solved yet. This **probably needs a more in depth discussion**. The current solution warns the user when duplicate bundle identifiers are found and tells them to explicitly create a map. This _could_ be improved in this pull request by showing an example mapping to put in the `Fastfile` - **thoughts??**

Below is an example of how things can get messed up...
```sh
Debug
	com.myapp.PizzaApp: Development Team
Debug-CookieApp
	com.myapp.CookieApp: Development Team
Release
	com.myapp.PizzaApp: match AppStore com.myapp.PizzaApp
Enterprise
	com.myapp.PizzaApp: In House
Release-CookieApp
	com.myapp.CookieApp: CookieApp AppStore
Enterprise-CookieApp
	com.myapp.CookieApp: In House
[22:56:42]: Detected provisioning profile mapping: {"com.myapp.PizzaApp"=>"In House", "com.myapp.CookieApp"=>"In House"}
```

You can see above that a map was created that resulted in "In House" which is _okay_ if I am doing an enterprise build but if I'm trying to do a release/appstore build, this same profile will get created and make the signing go 💥 

A **better** solution would be to pass which configuration is being used into the `CodeSigningMap` so that it can choose the correct bundle identifier and provisioning profile from above ^^ **Thoughts??**

### CODE REVIEW

File: gym/lib/gym/code_signing_mapping.rb [hunk 1]
```diff
@@ -107,23 +107,45 @@ def detect_project_profile_mapping
...
               end
             end
           end
+          
+          # Alerting user to explicitly specify a mapping it cannot be determined
+          unless bundle_identifiers_with_duplicates.empty?
+            UI.error("Couldn't automatically detect the provisioning profile mapping")
+            UI.error("There were multiple profiles for bundle identifier(s): #{bundle_identifiers_with_duplicates.uniq.join(', ')}")
+            UI.error("You need to provide an explicit mapping of what provisioning")
+            UI.error("profile to use for each bundle identifier of your app")
```
Comment (KrauseFx) [inline gym/lib/gym/code_signing_mapping.rb]: So we're not failing here, what do we do? Might be cool to inform the user of what profile we selected for this run, if we decide to continue running

File: gym/lib/gym/code_signing_mapping.rb [hunk 2]
```diff
@@ -107,23 +107,45 @@ def detect_project_profile_mapping
       self.project_paths.each do |project_path|
         UI.verbose("Parsing project file '#{project_path}' to find selected provisioning profiles")
 
+        # Storing bundle identifiers with duplicate profiles
+        # for informing user later on
+        bundle_identifiers_with_duplicates = []
```
Comment (KrauseFx) [inline gym/lib/gym/code_signing_mapping.rb]: One thing I noticed is that we set this to `[]` here, but we check if it's empty inside the `begin` and the loop? is that correct? How does it behave with multiple targets?

**Review by KrauseFx — APPROVED** [review]
Hey @joshdholtz, great work on this, those changes are super tricky, and really hard to implement and test. Oh, testing.. mhh, could we add 2 tests for this? ❤️ 

This looks great, thanks for your work on this 👍

File: gym/lib/gym/code_signing_mapping.rb (hunk 1 above)
Comment (joshdholtz) [inline gym/lib/gym/code_signing_mapping.rb]: It currently inform the user outside of this method somewhere what was chosen. It looks something like...
```sh
[00:14:24]: Couldn't automatically detect the provisioning profile mapping
[00:14:24]: There were multiple profiles for bundle identifier(s): com.snapifeye.SnapifeyePro, com.snapifeye.BrandedCam
[00:14:24]: You need to provide an explicit mapping of what provisioning
[00:14:24]: profile to use for each bundle identifier of your app
[00:14:24]: Detected provisioning profile mapping: {"com.myapp.PizzaApp"=>"In House", "com.myapp.CookieApp"=>"In House"}
```

I don't think I like the word "Detected" though 🤔

File: gym/lib/gym/code_signing_mapping.rb (hunk 2 above)
Comment (joshdholtz) [inline gym/lib/gym/code_signing_mapping.rb]: That was me being a dingus 😊 I moved them inside the begin. 

And I have not tested it with multiple targets but my example has 1 target with 6 configurations which will behave the same as 2 targets with 3 configurations (theoretically)... But I will test 😇

**Comment by joshdholtz** [comment]
@KrauseFx Aye aye, working on tests! Had to fix one cause the upgrade in `xcodeproj` actually failed the test cause the workspace had build errors so it couldn't resolve properly 😊 I think I got the test back to passing so will add new tests now 👌

**Comment by chefnobody** [comment]
Is it common for multiple Build Configurations (like: `Debug`, `Release`, `AdHoc`) within a Target (`OreoApp`) to each have _different_ Product Bundle Identifiers?

That seems strange, but I could be missing some obvious use-case for it.

**Comment by joshdholtz** [comment]
@chefnobody It would its common enough. I do it all the time 😇 Reasons being to have multiple apps for different environments installed at once (development, staging, production) and also if you have different white-labled/customer versions of an app

**Comment by chefnobody** [comment]
There are so many scenarios where this can go sideways its making my head spin... :-)

Question: If this comment is true: https://github.com/fastlane/fastlane/blob/master/gym/lib/gym/detect_values.rb#L59-L60 would it make more sense _today_ to take whatever the user provided as part of the `export_options.provisioningProfiles` and _not_ attempt to merge with any derived settings in Xcode (https://github.com/fastlane/fastlane/blob/master/gym/lib/gym/detect_values.rb#L66)? That way `gym` would just fail on archive if that mapping didn't line up with the organization of the Scheme's Targets, their Build Configurations and any Bundle Identifiers specified therein?

I ask because it seems strange to conflate the two mechanisms for setting a Provisioning Profile: One by letting the user specify a mapping when calling `gym` and another by trying to derive settings from Xcode.

Again, I could be missing a very valid and common use-case for needing to merge any derived settings.

(Appreciate your patience reading and responding to my questions)

**Comment by joshdholtz** [comment]
@KrauseFx Can you take a look at this again? Added in some logic (that @thasegaw suggested in the issue) that grabs the profile for the specified configuration.

File: gym/lib/gym/code_signing_mapping.rb [hunk 3]
```diff
@@ -103,26 +103,50 @@ def test_target?(build_settings)
...
       provisioning_profile_mapping = {}
+      specified_configuration = Gym.config[:configuration] || Gym.project.default_build_settings(key: "CONFIGURATION")
 
       self.project_paths.each do |project_path|
         UI.verbose("Parsing project file '#{project_path}' to find selected provisioning profiles")
+        UI.verbose("Finding provision profiles for '#{specified_configuration}'") if specified_configuration
 
         begin
+          # Storing bundle identifiers with duplicate profiles
+          # for informing user later on
```
Comment (taquitos) [inline gym/lib/gym/code_signing_mapping.rb]: best comment 🥅

File: gym/lib/gym/code_signing_mapping.rb [hunk 4]
```diff
@@ -103,26 +103,50 @@ def test_target?(build_settings)
...
+              # Creates the mapping for a bundle identifier and profile specifier/uuid
+              if has_profile_specifier
                 provisioning_profile_mapping[bundle_identifier] = provisioning_profile_specifier
-              elsif provisioning_profile_uuid.to_s.length > 0
+              elsif has_profile_uuid
                 provisioning_profile_mapping[bundle_identifier] = provisioning_profile_uuid
               end
             end
+
+            # Alerting user to explicitly specify a mapping it cannot be determined
```
Comment (taquitos) [inline gym/lib/gym/code_signing_mapping.rb]: > # Alerting user to explicitly specify a mapping it cannot be determined
 # Alerting user to explicitly specify a mapping if it cannot be determined

File: gym/spec/fixtures/projects/cocoapods/Example.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -36,8 +36,6 @@
 /* End PBXContainerItemProxy section */
 
 /* Begin PBXFileReference section */
-		7F17E2E4DB202336112485AA /* Pods-Example.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-Example.release.xcconfig"; path = "Pods/Target Support Files/Pods-Example/Pods-Example.release.xcconfig"; sourceTree = "<group>"; };
```
Comment (taquitos) [inline gym/spec/fixtures/projects/cocoapods/Example.xcodeproj/project.pbxproj]: are these just cocoapods updates?

**Review by taquitos — COMMENTED** [review]
A question and a typo, but otherwise 👍

**Comment by joshdholtz** [comment]
@taquitos That project had a reference that was deleted so the tests wouldn't build properly which was causing an issue with the `xcodeproj` upgrade since it resolves the configuration using external xcconfigs 😇

**Comment by taquitos** [comment]
Wooo thanks for fixing that project @joshdholtz!

**Comment by mathiasAichinger** [comment]
@joshdholtz I think this change is causing some red lines in our lane. We are using a xcconfig files hierarchy and variables to define the bundle identifier and the provisioning profile specifier. It seems neither the xcconfig includes and the variable names get resolved correctly by xcodeproj. I have already specified the export_options, but unfortunately this couldn't suppress the red lines. The build itself is working fine.

I have added some additional log output to print the target and the resolved variables.

`[15:51:36]: -----------------
[15:51:36]: --- Step: gym ---
[15:51:36]: -----------------
[15:51:37]: target: runtasticPRO
[15:51:37]: bundle_identifier: $(APPLICATION_BUNDLE_IDENTIFIER_AD_HOC_PREFIX)$(BASE_APPLICATION_BUNDLE_IDENTIFIER)
[15:51:37]: provisioning_profile_specifier: $($(BASE_APPLICATION_BUNDLE_IDENTIFIER)_PROFILE_SPECIFIER)
[15:51:37]: provisioning_profile_uuid:
[15:51:37]: target: runtasticLite
[15:51:37]: bundle_identifier: $(APPLICATION_BUNDLE_IDENTIFIER_AD_HOC_PREFIX)$(BASE_APPLICATION_BUNDLE_IDENTIFIER)
[15:51:37]: provisioning_profile_specifier: $($(BASE_APPLICATION_BUNDLE_IDENTIFIER)_PROFILE_SPECIFIER)
[15:51:37]: provisioning_profile_uuid:
[15:51:37]: Couldn't automatically detect the provisioning profile mapping
[15:51:37]: There were multiple profiles for bundle identifier(s): $(APPLICATION_BUNDLE_IDENTIFIER_AD_HOC_PREFIX)$(BASE_APPLICATION_BUNDLE_IDENTIFIER)
[15:51:37]: You need to provide an explicit mapping of what provisioning
[15:51:37]: profile to use for each bundle identifier of your app
[15:51:37]: target: WatchKitExtension
[15:51:37]: bundle_identifier:
[15:51:37]: provisioning_profile_specifier:
[15:51:37]: provisioning_profile_uuid:
[15:51:37]: Couldn't automatically detect the provisioning profile mapping
[15:51:37]: There were multiple profiles for bundle identifier(s): $(APPLICATION_BUNDLE_IDENTIFIER_AD_HOC_PREFIX)$(BASE_APPLICATION_BUNDLE_IDENTIFIER)
[15:51:37]: You need to provide an explicit mapping of what provisioning
[15:51:37]: profile to use for each bundle identifier of your app
[15:51:37]: target: WatchKitExtensionLite
[15:51:37]: bundle_identifier:
[15:51:37]: provisioning_profile_specifier:
[15:51:37]: provisioning_profile_uuid:
[15:51:37]: Couldn't automatically detect the provisioning profile mapping
[15:51:37]: There were multiple profiles for bundle identifier(s): $(APPLICATION_BUNDLE_IDENTIFIER_AD_HOC_PREFIX)$(BASE_APPLICATION_BUNDLE_IDENTIFIER)
[15:51:37]: You need to provide an explicit mapping of what provisioning
[15:51:37]: profile to use for each bundle identifier of your app
[15:51:37]: target: WatchApp
[15:51:37]: bundle_identifier:
[15:51:37]: provisioning_profile_specifier: $(at.runtastic.runtastic.pro.watchapp_PROFILE_SPECIFIER)
[15:51:37]: provisioning_profile_uuid:
[15:51:37]: Couldn't automatically detect the provisioning profile mapping
[15:51:37]: There were multiple profiles for bundle identifier(s): $(APPLICATION_BUNDLE_IDENTIFIER_AD_HOC_PREFIX)$(BASE_APPLICATION_BUNDLE_IDENTIFIER)
[15:51:37]: You need to provide an explicit mapping of what provisioning
[15:51:37]: profile to use for each bundle identifier of your app
[15:51:37]: target: WatchAppLite
[15:51:37]: bundle_identifier:
[15:51:37]: provisioning_profile_specifier: $(at.runtastic.gpssportapp.watchapp_PROFILE_SPECIFIER)
[15:51:37]: provisioning_profile_uuid:
[15:51:37]: Couldn't automatically detect the provisioning profile mapping
[15:51:37]: There were multiple profiles for bundle identifier(s): $(APPLICATION_BUNDLE_IDENTIFIER_AD_HOC_PREFIX)$(BASE_APPLICATION_BUNDLE_IDENTIFIER),
[15:51:37]: You need to provide an explicit mapping of what provisioning
[15:51:37]: profile to use for each bundle identifier of your app`

**Comment by KrauseFx** [comment]
@mathiasAichinger Please submit a new issue instead of commenting on old PRs, thanks

**Pull request merged**